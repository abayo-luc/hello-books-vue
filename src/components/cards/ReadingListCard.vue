<template >
  <div class="reading-list-card dark-card">
    <div class="single-item" v-for="(book, index) in books" :key="index">
      <img src="../../assets/icons/open-book.svg" class="icon" />
      <div class="book-details">
        <p class="book-name">Things Falll Apart</p>
        <p class="book-pages">
          <span>Page &#8470; :</span>1232
        </p>
      </div>
    </div>
  </div>
</template>
<script>
export default {
  name: 'ReadingListCard',
  data() {
    return {
      books: [...Array(5).keys()]
    };
  }
};
</script>
<style lang="scss" scoped>
.reading-list-card {
  position: absolute;
  top: 4em;
  border-radius: 20px;
  width: 100%;
  padding: 10px;
  min-height: 10em;
  box-shadow: 0 4px 8px 0 rgba(35, 33, 38, 0.1);
  transition: 0.3s;
  .icon {
    height: 25px;
    margin-right: 10px;
  }
  .single-item,
  .ic-book {
    display: flex;
    flex-direction: row;
  }
  .single-item {
    justify-content: flex-start;
    align-items: center;
    margin: 7.5px 0px;
    p.book-pages {
      font-size: 11px;
      font-style: italic;
      color: #f4f6f8;
      margin-left: 3px;
    }
  }
}
</style>
