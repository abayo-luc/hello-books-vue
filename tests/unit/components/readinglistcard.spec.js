import {
  mount
} from '@vue/test-utils';
import ReadingListCard from '../../../src/components/cards/ReadingListCard.vue';

const wrapper = mount(ReadingListCard);

describe('ReadingListCard.vue', () => {
  it('should match snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
