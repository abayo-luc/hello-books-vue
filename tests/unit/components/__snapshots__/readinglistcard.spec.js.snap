// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ReadingListCard.vue should match snapshot 1`] = `
<div class="reading-list-card dark-card">
  <div class="single-item"><img src="../../assets/icons/open-book.svg" class="icon">
    <div class="book-details">
      <p class="book-name">Things Falll Apart</p>
      <p class="book-pages"><span>Page № :</span>1232
      </p>
    </div>
  </div>
  <div class="single-item"><img src="../../assets/icons/open-book.svg" class="icon">
    <div class="book-details">
      <p class="book-name">Things Falll Apart</p>
      <p class="book-pages"><span>Page № :</span>1232
      </p>
    </div>
  </div>
  <div class="single-item"><img src="../../assets/icons/open-book.svg" class="icon">
    <div class="book-details">
      <p class="book-name">Things Falll Apart</p>
      <p class="book-pages"><span>Page № :</span>1232
      </p>
    </div>
  </div>
  <div class="single-item"><img src="../../assets/icons/open-book.svg" class="icon">
    <div class="book-details">
      <p class="book-name">Things Falll Apart</p>
      <p class="book-pages"><span>Page № :</span>1232
      </p>
    </div>
  </div>
  <div class="single-item"><img src="../../assets/icons/open-book.svg" class="icon">
    <div class="book-details">
      <p class="book-name">Things Falll Apart</p>
      <p class="book-pages"><span>Page № :</span>1232
      </p>
    </div>
  </div>
</div>
`;
